src/executor/evaluate/expr.rs

```diff
@@ -95,31 +97,22 @@ pub fn case<'a>(
     when_then: Vec<(Evaluated<'a>, Evaluated<'a>)>,
     else_result: Option<Evaluated<'a>>,
 ) -> Result<Evaluated<'a>> {
+    use std::mem::discriminant as disc;
+
     let (_, then_results) = when_then.iter().cloned().unzip::<_, _, Vec<_>, Vec<_>>();
 
     let results = match &else_result {
         Some(result) => [&then_results[..], &[result.to_owned()]].concat(),
         None => then_results,
     };
 
-    let result_types = results
+    let type_diff = results
         .iter()
-        .map(|rt| -> Result<i32> {
-            match rt.to_owned().try_into()? {
-                Value::Bool(_) => Ok(1),
-                Value::Interval(_) => Ok(2),
-                Value::Date(_) => Ok(3),
-                Value::I64(_) => Ok(4),
-                Value::F64(_) => Ok(5),
-                Value::Str(_) => Ok(6),
-                Value::Time(_) => Ok(7),
-                Value::Timestamp(_) => Ok(8),
-                Value::Null => Ok(9),
-            }
-        })
-        .collect::<Vec<_>>();
+        .map(|r| -> Value { r.to_owned().try_into().unwrap() })
+        .tuple_windows()
+        .any(|(left, right)| disc(&left) != disc(&right));
```

Comment:
@panarch 
I've simplified this part with a function called `std::mem::discriminant()`, which returns a variant of Enum. 
And I also used `tuple_windows()` which I found out while I was searching about the `windows()` function you recommended me to use on recent feedback. The basic functionality of those are simillar.
Thankfully, those aweful `Value::Variant(_) => Ok(i32)`s are gone now! 

Tell me if this is a good idea, or should I try another better method. Learning a lot from this issue! 😄